### PR TITLE
194 pass real reflectivity and sld data from the project method exposing lib data

### PR DESCRIPTION
**Pull request by andped10**
It is now possible to get the reflectivity and sld profile from the lib.

project now have a default minimizer and calculator

Added unique names various places

### CODE REVIEW

File: src/easyreflectometry/calculators/calculator_base.py [hunk 1]
```diff
@@ -158,14 +158,22 @@ def remove_item_from_model(self, item_id: str, model_id: str) -> None:
...
+    def reflectity_profile(self, x_array: np.ndarray, model_id: str) -> np.ndarray:
         """Function to perform a fit.
 
         :param x_array: points to be calculated at
         :param model_id: The model id
         """
         return self._wrapper.calculate(x_array, model_id)
 
+    def fit_func(self, x_array: np.ndarray, model_id: str) -> np.ndarray:
+        """Function to perform a fit.
```
Comment (rozyczko) [inline src/easyreflectometry/calculators/calculator_base.py:170]: This does not actually perform a fit, does it?
It only calculates the profile based on the current set of parameters in the model.

File: src/easyreflectometry/project.py [hunk 2]
```diff
@@ -49,8 +41,8 @@ def __init__(self):
         self._path_project_parent = Path(os.path.expanduser('~'))
         self._models = ModelCollection(populate_if_none=False, unique_name='project_models')
         self._materials = MaterialCollection(populate_if_none=False, unique_name='project_materials')
-        self._calculator = None
-        self._minimizer: AvailableMinimizers = None
+        self._calculator = CalculatorFactory()
+        self._minimizer = AvailableMinimizers.LMFit_leastsq
```
Comment (rozyczko) [inline src/easyreflectometry/project.py:45]: Maybe the default minimizer should be defined somewhere central? Top of the file as `DEFAULT_MINIMIZER`?
So you can avoid duplication and help with maintenance.

File: src/easyreflectometry/project.py [hunk 3]
```diff
@@ -132,20 +146,21 @@ def default_model(self):
         self._replace_collection(MaterialCollection(), self._materials)
 
         layers = [
-            Layer(material=self._materials[0], thickness=0.0, roughness=0.0, name='Vacuum Layer'),
-            Layer(material=self._materials[1], thickness=100.0, roughness=3.0, name='Multi-layer'),
-            Layer(material=self._materials[2], thickness=0.0, roughness=1.2, name='Si Layer'),
+            Layer(material=self._materials[0], thickness=0.0, roughness=0.0, name='Vacuum Layer', interface=self._calculator),
```
Comment (rozyczko) [inline src/easyreflectometry/project.py:149]: Maybe you can make `CalculatorFactory` a singleton, so you don't carry this with every call, to Layer, Multilayer, Sample etc.?
Do you need more than one instance of this factory?

File: tests/test_project.py [hunk 4]
```diff
@@ -109,6 +111,51 @@ def test_default_model(self):
...
+        # When
+        project = Project()
+        project.default_model()
+
+        # Then
+        sample_data = project.sld_data_for_model_at_index(0)
+
+        # Expect
+        assert len(sample_data.x) == 500
+        assert_almost_equal(
```
Comment (rozyczko) [inline tests/test_project.py:124]: It is preferred to use `assert_allclose`, which has improved features:

https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_almost_equal.html

**Review by rozyczko — APPROVED** [review]
Looks good, minor suggestions added.

File: src/easyreflectometry/calculators/calculator_base.py (hunk 1 above)
Comment (andped10) [reply src/easyreflectometry/calculators/calculator_base.py]: removed fit_function from calculator_base.  Adjusted docstring

File: src/easyreflectometry/project.py (hunk 2 above)
Comment (andped10) [reply src/easyreflectometry/project.py:45]: Good point. Changed

File: src/easyreflectometry/project.py (hunk 3 above)
Comment (andped10) [reply src/easyreflectometry/project.py:151]: You are most certainly right.  I have not yet fully understood how when we need to pass the factory and I decided to be very explicit to be sure that it hits all objects.

I would like to make the suggested change, but that will have to wait to another PR

File: tests/test_project.py (hunk 4 above)
Comment (andped10) [reply tests/test_project.py]: changed assert in tests

**Pull request merged**